Slingshot.Api/Abstract/Repository.cs

```diff
@@ -85,7 +85,8 @@ public virtual string RepositoryName
                 {
                     if (_inputUri.Segments.Length > 2)
                     {
-                        _repositoryName = _inputUri.Segments[2].Trim(Constants.Path.SlashChars);
+                        // remove query string and slash at the end
+                        _repositoryName = _inputUri.Segments[2].Split(Constants.Path.QuestionChars)[0].Split(Constants.Path.SlashChars)[0];
```

Comment:
It would help understanding to have an example of what we're parsing in the comment, and what we're trying to get out of it.

Also, `Constants.Path.QuestionChars` seems overkills and makes the code longer/trickier to understand than simply using '?'.